proposals/2020-07-01_aws-kms-keyring-redesign/change.md

```diff
@@ -0,0 +1,327 @@
+
+**Behavior**
+
+The AWS KMS symmetric region discovery keyring MUST...
+
+1. Be configured with a single AWS SDK client.
+   1. Determine the AWS region associated with the provided AWS SDK client.
+      In some implementations,
+      this MAY require an additional AWS region argument
+      during AWS KMS symmetric region discovery keyring initialization.
+      If the AWS SDK returns the AWS SDK client’s region
+      directly from the AWS SDK client
+      in an intuitive manner (by using a single function/method call),
+      implementations MUST NOT require an additional AWS region argument.
+   2. Only communicate with the AWS region associated with the AWS SDK client.
+2. Use the AWS SDK client provided at initialization for all AWS KMS API calls.
+3. MUST return an error for the OnEncrypt keyring API, rather than returning the unchanged encryption material.
+4. Provide the encrypted data key’s [provider info](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/structures.md#key-provider-information)
+   as part of the [AWS KMS Decrypt API](https://docs.aws.amazon.com/kms/latest/APIReference/API_Decrypt.html) call.
+
+The AWS KMS symmetric region discovery keyring MAY...
+
+1. Be configured with an AWS account ID.
+   1. If an AWS account ID is provided,
+      the AWS KMS symmetric region discovery keyring MUST only decrypt encrypted data keys
+      that were encrypted using an AWS KMS CMK in that AWS account, for the keyring’s region.
+   2. If no AWS account ID is provided,
+      the AWS KMS symmetric region discovery keyring MUST _attempt_ to decrypt all encrypted data keys in the keyring’s region.
+2. Be configured with a list of string [grant tokens](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md#grant-tokens)
+   to be included in all AWS KMS calls.
+
+**Decryption Contract**
+
+1. The keyring MUST have access to an AWS KMS client that is configured with credentials for an AWS principal
+   that passes an [AWS KMS Decrypt API](https://docs.aws.amazon.com/kms/latest/APIReference/API_Decrypt.html)
+   authorization check for the encrypted data key.
+2. The keyring is unable to write a decryption contract,
+   but it MAY fulfill any AWS KMS decryption contract for the configured AWS region and, if provided, the AWS account ID.
+   1. Fulfilling any AWS KMS decryption contract MUST be sufficient to succeed.
+
+**Initialization**
+
+On keyring initialization, an AWS KMS symmetric region discovery keyring MUST define the following:
+
+- AWS SDK client (and the AWS SDK client’s region)
+
+On keyring initialization, an AWS KMS symmetric region discovery keyring MAY define the following:
+
+- AWS account ID
+- Grant Tokens
+
+### Keyring-Producing Operations
+
+**Please note the following names are not finalized.**
+
+The ESDK MUST provide customers a way to use the following:
```

Comment:
```suggestion
The ESDK MUST provide customers a way to initialize the following:
```

Or you could say "MUST provide keyring-producing operations for the following". The concept of "keyring-producing operation" is something we should introduce at a higher level to clarify how that relates to all the Initialization sections.

Reply:
I already introduce the concept higher up, but will change to initialize here (though I will most likely rephrase this for the "derived keyring" change (or other naming change).